widget/card.go

```diff
@@ -0,0 +1,204 @@
+		return fyne.NewSize(c.card.Image.MinSize().Width+theme.Padding(), cardMediaHeight+theme.Padding())
+	}
+
+	min := fyne.NewSize(theme.Padding()*5, theme.Padding()*5) // content padding plus 1 pad border
+	if hasImage {
+		min = fyne.NewSize(min.Width, min.Height+cardMediaHeight)
+	}
+
+	if hasHeader {
+		min = fyne.NewSize(fyne.Max(min.Width, c.header.MinSize().Width+theme.Padding()*5),
+			min.Height+c.header.MinSize().Height)
+		if hasSubHeader || hasContent {
+			min.Height += theme.Padding()
+		}
+	}
+	if hasSubHeader {
+		min = fyne.NewSize(fyne.Max(min.Width, c.subHeader.MinSize().Width+theme.Padding()*5),
+			min.Height+c.subHeader.MinSize().Height)
+		if hasContent {
+			min.Height += theme.Padding()
+		}
+	}
+	if hasContent {
+		min = fyne.NewSize(fyne.Max(min.Width, c.card.Content.MinSize().Width+theme.Padding()*3),
+			min.Height+c.card.Content.MinSize().Height)
+	}
+
+	return min
+}
+
+// Layout the components of the card container.
+func (c *cardRenderer) Layout(size fyne.Size) {
+	pos := fyne.NewPos(theme.Padding()/2, theme.Padding()/2)
+	size = size.Subtract(fyne.NewSize(theme.Padding(), theme.Padding()))
+	c.LayoutShadow(size, pos)
+
+	if c.card.Image != nil {
+		c.card.Image.Move(pos)
+		c.card.Image.Resize(fyne.NewSize(size.Width, cardMediaHeight))
+		pos.Y += cardMediaHeight
+	}
+
+	size.Width -= theme.Padding() * 4
+	pos.X += theme.Padding() * 2
+	pos.Y += theme.Padding()
+
+	if c.card.Title != "" {
+		height := c.header.MinSize().Height
+		c.header.Move(pos)
+		c.header.Resize(fyne.NewSize(size.Width, height))
+		pos.Y += height + theme.Padding()
+	}
+
+	if c.card.SubTitle != "" {
+		height := c.subHeader.MinSize().Height
+		c.subHeader.Move(pos)
+		c.subHeader.Resize(fyne.NewSize(size.Width, height))
+		pos.Y += height + theme.Padding()
+	}
+
+	size.Width += theme.Padding() * 2
+	pos.X -= theme.Padding()
+	pos.Y += theme.Padding()
+
+	if c.card.Content != nil {
+		height := size.Height - pos.Y - theme.Padding()*3
+		c.card.Content.Move(pos)
+		c.card.Content.Resize(fyne.NewSize(size.Width, height))
+	}
+}
+
+// applyTheme updates this button to match the current theme
+func (c *cardRenderer) applyTheme() {
+	if c.header != nil {
+		c.header.TextSize = int(float32(theme.TextSize()) * 1.7)
+		c.header.Color = theme.TextColor()
+	}
+	if c.subHeader != nil {
+		c.subHeader.TextSize = theme.TextSize()
+		c.subHeader.Color = theme.TextColor()
+	}
+}
+
+func (c *cardRenderer) BackgroundColor() color.Color {
+	return theme.BackgroundColor()
+}
+
+func (c *cardRenderer) Refresh() {
+	c.header.Text = c.card.Title
+	c.header.Refresh()
+	c.subHeader.Text = c.card.SubTitle
+	c.subHeader.Refresh()
+
+	c.applyTheme()
+	c.Layout(c.card.Size())
+	canvas.Refresh(c.card.super())
+}
+
+// CardContainer widget groups title, subtitle with content and a header image
+type CardContainer struct {
+	BaseWidget
+	Title, SubTitle string
+	Image           *canvas.Image
+	Content         fyne.CanvasObject
```

Comment:
I can haz two images?

From the spec:
> Media [optional]
> Cards can include a variety of media, including photos, and graphics, such as weather icons.

Also what about the rest of the spec: Thumbnail, Supporting Text, Buttons, Icons? This implementation feels very restrictive.

Reply:
Sorry but I do not understand the question - are you asking if two images can be added, or suggesting that it's strange that they can?

The supporting text is, in this implementation, the "Content" field and it can be anything you like.
Buttons were addressed in the PR opening: "We can add more to this later, such as the buttons (style challenges as they are depicted like hyperlink...?)" - I figured that being part of the button bar we would address icons at the same time.
Thumbnail is presented as an image clipped to a circle - as we cannot do this yet I opted to leave it out.

This is indeed not a "full card" implementation, as noted in the body of the PR.
It is intended as a starting point that is at least better than group, which as a poor API and needs to be deprecated.